astropy/table/column.py

```diff
@@ -214,6 +213,8 @@ def __new__(cls, data=None, name=None,
                 meta = deepcopy(data.info.meta)
 
         else:
+            if not six.PY2 and np.dtype(dtype).char == 'S':
+                data = cls._encode_str(data)
```

Comment:
If I understand correctly the `dtype` variable here can be not None only if a dtype was explicitly set when creating a table (`Table(..., dtype=['S'])`), as in your tests and doc examples. So all this does not apply when creating a table/column from a numpy array (or FITS recarray) without specifying the dtype. 
I think this relates to the first point from @astrofrog , but I just wanted to be sure it is intended ?
(And maybe a comment explaining this would be useful for the future ?)